File: src/main/java/com/christophsens/cvenhancer/exceptions/OpenAiExceptionHandler.java
package com.christophsens.cvenhancer.exceptions;

import org.springframework.http.ProblemDetail;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.server.ResponseStatusException;

@RestControllerAdvice
public class OpenAiExceptionHandler {

    @ExceptionHandler(ResponseStatusException.class)
    ProblemDetail handleResponseStatusException(ResponseStatusException ex) {
        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(ex.getStatusCode(), ex.getReason());
        problemDetail.setTitle(ex.getStatusCode().toString());
        return problemDetail;
    }
}

File: src/main/kotlin/com/christophsens/cvenhancer/exceptions/OpenAiExceptionHandler.kt
package com.christophsens.cvenhancer.exceptions

import org.springframework.http.ProblemDetail
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.server.ResponseStatusException

@RestControllerAdvice
class OpenAiExceptionHandler {

    @ExceptionHandler(ResponseStatusException::class)
    fun handleResponseStatusException(ex: ResponseStatusException): ProblemDetail {
        val problemDetail = ProblemDetail.forStatusAndDetail(ex.statusCode, ex.reason)
        problemDetail.title = ex.statusCode.toString()
        return problemDetail
    }
}